genie-web/src/test/groovy/com/netflix/genie/web/tasks/job/JobCompletionServiceSpec.groovy

```diff
@@ -185,11 +180,53 @@ class JobCompletionServiceSpec extends Specification {
         1 * jobPersistenceService.updateJobStatus(jobId, _, _)
         1 * mailService.sendEmail('admin@netflix.com', _, _)
         1 * completionTimer.record(_, TimeUnit.NANOSECONDS)
-        timerTagsCapture == ImmutableMap.of(
-                MetricsConstants.TagKeys.STATUS, MetricsConstants.TagValues.SUCCESS,
-                JobCompletionService.JOB_FINAL_STATE, JobStatus.FAILED.toString()
-        )
+        verifySuccessTags(timerTagsCapture) \
+         && timerTagsCapture.get(JobCompletionService.JOB_FINAL_STATE).equals(JobStatus.FAILED.toString())
         3 * errorCounter.increment()
+
+        when:
+        jobCompletionService.handleJobCompletion(new JobFinishedEvent(jobId, JobFinishedReason.PROCESS_COMPLETED, "null", this))
+        then:
+        1 * jobSearchService.getJob(jobId) >> new Job.Builder(NAME, USER, VERSION)
```

Comment:
You are reusing identical `Command` and `Cluster`, creating them once in the `setup:` section would make this test easier to read and maintain